pkg/admission/namespacelifecycle/admission.go

```diff
@@ -144,7 +144,10 @@ func (l *workspaceNamespaceLifecycle) SetExternalKubeClientSet(client kubernetes
 
 func (l *workspaceNamespaceLifecycle) SetKcpInformers(informers kcpinformers.SharedInformerFactory) {
 	l.SetReadyFunc(informers.Tenancy().V1alpha1().ClusterWorkspaces().Informer().HasSynced)
-	l.workspaceLister = informers.Tenancy().V1alpha1().ClusterWorkspaces().Lister()
+
+	l.getClusterWorkspace = func(clusterName logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspace, error) {
+		return informers.Tenancy().V1alpha1().ClusterWorkspaces().Lister().Get(kcpcache.ToClusterAwareKey(clusterName.String(), "", name))
```

Comment:
Sorry, this is a *different* `ToClusterAwareKey` - you'll still want the old one, or we can temporarily just `+ "|" +` on these

Reply:
Er I guess without a namespace, it's fine